### ISSUE TITLE
Saving email contents in wallabag

### ISSUE THREAD

**Issue by burkemw3** [wallabag/wallabag#3746]
I have some email newsletters I subscribe to that I subscribe to. I'd like to be able to read these in wallabag, where I'm reading similar content.

Creating a new issue, since I misunderstood #239. This may be a possible workaround for #2229. I am lifting some comments from #239 into this issue.

- @Maintainers, are you receptive to a new endpoint for a mail service posting message bodies to be saved?
- I have a PoC of this already. Would like to mainline it, especially so others can use, but will maintain separately if necessary
- One challenge is what URL should be used for the entry?

---

After the email processor is developed, I could see it being used to solve #239 as well. I could even imagine "site filters" for email. There'd be some from/to/subject matching, and could be a body tag for capturing the email content, or a url tag for fetching a URL from the message.

**Comment by sschuldenzucker** [comment]
Hi, any news on this? This is the only reason I stick to instapaper…

**Comment by j0k3r** [comment]
I think it's available on wallabag.it? Poke @nicosomb

**Comment by sschuldenzucker** [comment]
No, they only allow saving links from email, but not contents afaiu.
I can understand why, but in this strange age of newsletters, saving
contents is just super useful.

**Comment by harmtemolder** [comment]
I built this for myself, thought I’d share: https://git.sr.ht/~harmtemolder/wallamail.

It’s still pretty much alpha, but I have it working on an Ubuntu VPS through Docker and Traefik. It’s basically just an HTTP endpoint that accepts JSON from CloudMailin (but could be tweaked for similar services) and forwards the email contents to the wallabag API. I now use my CloudMailin address to subscribe to newsletters and can read them in wallabag.

**Comment by ekasprzak** [comment]
For anyone interested, I found this nice tool: 
https://www.publishthis.email

You just send an email with any content to provided email there and it generates that content as a publicly available web page, so it can be easily wallabaged. 
For newsletters etc it works great, the only inconvenience is that you need to forward emails manually (at least in gmail it's not possible to setup auto-forward rule if you don't have access to the other mailbox).

**Comment by jeff47** [comment]
> https://www.publishthis.email

Sadly it looks dead now...

**Comment by ekasprzak** [comment]
> > https://www.publishthis.email
> 
> Sadly it looks dead now...

Yeah, that's a shame, cause it was a perfect tool for the job.
I found some alternative, not as easy and convenient as above, but it works.

Some time ago I discovered that wallabag extension has this great option - "Content fetching from the browser".
After that, I was looking for a "notepad like" webpage, where I can just copy content from my newsletter, paste it into that website and then use wallabag "fetch content from the browser" feature.
Unfortunately, this option is not perfect, on same pages it works great, on many others it collects so much garbage content which makes it useless on many web pages.

But, after a lot of research I found this one:
https://quicknote.io/

wallabag extension works great with that web page. Yes, it's less convenient, you need to take more steps (including changing a title if you want nice-looking article), but in the end it does what I want, so I recommend you to check it.

If anyone has found some better/easier alternative, please let us know;)

**Comment by HolgerAusB** [comment]
For selfhosters, [Roundcube](https://github.com/roundcube/roundcubemail) could be interesting. Wallabagger is taking the content without hassle. I think there is just a bit of fine tuning necessary with a special site_config to prevent gaps. (Just tested it quick with one newsletter)

Roundcube is not a full mail server it is just a web frontend for imap/smtp
